packages/amplify_api/lib/graphql/graphql_subscription_event.dart

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_api/amplify_api.dart';
+
+/// GraphQL subscription event types.
+enum GraphQLSubscriptionEventType {
+  /// Triggered when data is received from a GraphQL subscription.
+  data,
+
+  /// Triggered when a GraphQL subscription completes either due to cancellation
+  /// or a platform error.
+  done,
+
+  /// Triggered when a platform error occurs for a subscription.
+  ///
+  /// Currently only valid in Dart.
+  error,
+}
+
+/// Helper functions for [GraphQLSubscriptionEventType].
+extension GraphQLSubscriptionEventTypeX on GraphQLSubscriptionEventType {
```

Comment:
Minor nit: naming seems strange here with "TypeX" 

Should name match function more like EventTypeStringParser or GraphQLSubscriptionEventTypeExtension?

Reply:
Hmm - could just be a personal preference thing. I've always liked `TypeX` for extensions. Typically, they aren't referred to by name, only when there are static methods associated with them like in this case. 

For a public type which is referred to by name, I agree there are probably more descriptive names/approaches. This one isn't exported, though, it's only an internal helper.